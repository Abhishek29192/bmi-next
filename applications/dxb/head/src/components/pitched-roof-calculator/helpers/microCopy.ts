export { MicroCopy, MicroCopyContext, getMicroCopy } from "@bmi/components";
