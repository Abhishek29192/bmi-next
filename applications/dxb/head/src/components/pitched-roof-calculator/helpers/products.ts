import {
  createProduct as createESProduct,
  Product as ESProduct,
  ProductReference
} from "@bmi/elasticsearch-types";
import { convertStrToBool } from "../../../utils/convertStrToBool";
import groupBy from "../../../utils/groupBy";
import {
  Accessory,
  Data,
  GutterHook,
  GutterVariant,
  LengthBasedProduct,
  lengthBasedProducts,
  multiValueProducts,
  NestedProductReferences,
  ProductType,
  ReferencedTileProducts,
  RidgeOption,
  Tile,
  Underlay,
  VergeOption,
  WidthBasedProduct,
  widthBasedProducts
} from "../types/v2";

// Function overloads to make it working correctly with products of different types
export function transformClassificationAttributes(
  products: ESProduct[],
  productType: ProductType.tile
): Tile[];
export function transformClassificationAttributes(
  products: ESProduct[],
  productType: ProductType.underlay
): Underlay[];
export function transformClassificationAttributes(
  products: ESProduct[],
  productType: ProductType.gutterHook | ProductType.gutter
): Array<GutterHook | GutterVariant>;
export function transformClassificationAttributes(
  products: ESProduct[]
): (RidgeOption | LengthBasedProduct | WidthBasedProduct | Accessory)[];
export function transformClassificationAttributes(
  products: ESProduct[],
  productType?: ProductType
) {
  if (!products) {
    return [];
  }

  return products
    .map((product) => {
      const {
        "APPEARANCEATTRIBUTES.COLOUR": color,
        "GENERALINFORMATION.CLASSIFICATION": category,
        "TILESATTRIBUTES.BROKENBOND": brokenBond,
        "MEASUREMENTS.WIDTH": width,
        "TILESATTRIBUTES.MINIMUMBATTENSPACING": minBattenSpacing,
        "TILESATTRIBUTES.MAXIMUMBATTENSPACING": maxBattenSpacing,
        "TILESATTRIBUTES.RIDGESPACE": ridgeSpacing,
        "TILESATTRIBUTES.EAVEGAUGE": eaveGauge,
        "MEASUREMENTS.LENGTH": length,
        "UNDERLAYATTRIBUTES.OVERLAP": overlap,
        "PACKAGINGINFORMATION.QUANTITYPERUNIT": packSize,
        ...rest
      } = product;

      // Code in catch block will be executed if the value expected to be number isn't a number
      try {
        const initialData = {
          ...rest,
          packSize: Number(packSize?.[0].value) || 1
        };

        if (productType === ProductType.tile) {
          return {
            ...initialData,
            brokenBond: convertStrToBool(brokenBond?.[0].value),
            category: category[0].name,
            color: color[0].name,
            width: convertToCentimeters(width[0]),
            length: convertToCentimeters(length[0]),
            minBattenSpacing: convertToCentimeters(minBattenSpacing[0]),
            maxBattenSpacing: convertToCentimeters(maxBattenSpacing[0]),
            ridgeSpacing: convertToCentimeters(ridgeSpacing[0]),
            eaveGauge: convertToCentimeters(eaveGauge[0])
          };
        }

        if (productType === ProductType.underlay) {
          return {
            ...initialData,
            overlap: convertToCentimeters(overlap[0]),
            width: convertToCentimeters(width[0]),
            length: convertToCentimeters(length[0])
          };
        }

        if (
          productType === ProductType.gutter ||
          productType === ProductType.gutterHook
        ) {
          return {
            ...initialData,
            length: convertToCentimeters(length[0])
          };
        }

        return {
          ...initialData,
          category: category?.[0].name,
          length: convertToCentimeters(length?.[0], false),
          width: convertToCentimeters(width?.[0], false)
        };
      } catch (err) {
        return null;
      }
    })
    .filter(Boolean);
}

export const groupByProductType = (
  products: ESProduct[]
): {
  tiles: Tile[];
  underlays: Underlay[];
  gutters: GutterVariant[];
  gutterHooks: GutterHook[];
} => {
  const groupedProducts = groupBy(
    products,
    (product) => product["GENERALINFORMATION.PRODUCTTYPE"][0].code
  );

  return {
    tiles: transformClassificationAttributes(
      groupedProducts.MAIN_TILE,
      ProductType.tile
    ),
    underlays: transformClassificationAttributes(
      groupedProducts.UNDERLAY,
      ProductType.underlay
    ),
    gutters: transformClassificationAttributes(
      groupedProducts.GUTTER,
      ProductType.gutter
    ),
    gutterHooks: transformClassificationAttributes(
      groupedProducts.GUTTER_HOOK,
      ProductType.gutterHook
    )
  };
};

export const prepareProducts = (products: ESProduct[]): Data => {
  const transformedProducts = groupByProductType(products);

  const groupedTiles = groupBy(
    transformedProducts.tiles,
    (tile) => tile.baseProduct.code
  );

  const groupedGutters = groupBy(
    transformedProducts.gutters,
    (gutter) => gutter.baseProduct.code
  );

  return {
    tiles: groupedTiles,
    gutters: groupedGutters,
    gutterHooks: transformedProducts.gutterHooks,
    underlays: transformedProducts.underlays
  };
};

export const convertToCentimeters = (
  input: {
    value: string;
    code: string;
    name: string;
  },
  throwOnTypeError = true
): number => {
  const numberValue = Number(input?.value);

  if (!numberValue) {
    if (throwOnTypeError) {
      throw new Error("Value is not a number");
    }

    return null;
  }

  const { code, value } = input;

  //cm, mm, m, μm
  const unitOfLength = code.slice(value.length).toLowerCase();
  switch (unitOfLength) {
    case "mm":
      return numberValue / 10;
    case "m":
      return numberValue * 100;
    case "μm":
      return numberValue / 10000;
    default:
      return numberValue;
  }
};

const getProductsByReference = (
  productReferences: ProductReference[],
  products: ESProduct[]
) =>
  productReferences
    .map((productReference) => {
      const product = products.find(
        (product) => product.code === productReference.code
      );

      if (!product) {
        return null;
      }

      const transformedProduct = transformClassificationAttributes([
        product
      ])[0];

      const isLengthBasedProduct = lengthBasedProducts.includes(
        productReference.type
      );
      const isWidthBasedProduct = widthBasedProducts.includes(
        productReference.type
      );

      if (isLengthBasedProduct && !transformedProduct.length) {
        return null;
      }

      if (isWidthBasedProduct && !transformedProduct.width) {
        return null;
      }

      return {
        referenceType: productReference.type,
        product: transformedProduct
      };
    })
    .filter(Boolean);

export const transformProductReferences = <
  T = NestedProductReferences | ReferencedTileProducts
>(
  productReferences: ProductReference[],
  products: ESProduct[],
  fieldsMapper: Record<string, string>
): Partial<T> => {
  const groupedReferencesProducts = getProductsByReference(
    productReferences,
    products
  );
  const references = Object.keys(fieldsMapper);
  return references.reduce((prev, productReference) => {
    const isArray = multiValueProducts.includes(productReference);
    // eslint-disable-next-line security/detect-object-injection
    const key = fieldsMapper[productReference];

    return {
      ...prev,
      [key]: isArray
        ? groupedReferencesProducts
            .filter(({ referenceType }) => referenceType === productReference)
            .map(({ product }) => product)
        : groupedReferencesProducts.find(
            ({ referenceType }) => referenceType === productReference
          )?.product
    };
  }, {});
};

export const getVergeOption = (
  productReferences: Partial<VergeOption>
): VergeOption | undefined => {
  const { left, right, rightStart, leftStart, halfLeft, halfRight } =
    productReferences;
  if (!left || !right) {
    return undefined;
  }

  return {
    left,
    right,
    rightStart,
    leftStart,
    halfLeft,
    halfRight
  };
};

export const createProduct = <T extends ESProduct>(data: Partial<T>): T =>
  ({
    ...createESProduct(),
    ...data
  } as T);
