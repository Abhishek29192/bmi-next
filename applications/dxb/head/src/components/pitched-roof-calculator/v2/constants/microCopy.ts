export const microCopy = {
  ROOF_SELECTION_TITLE: "roofSelection.title",
  ROOF_SELECTION_SUBTITLE: "roofSelection.subtitle",
  ROOF_SELECTION_GABLE: "roofSelection.gable",
  ROOF_SELECTION_HIPPED: "roofSelection.hipped",
  ROOF_SELECTION_SLOPED: "roofSelection.sloped",
  ROOF_SELECTION_ROOF: "roofSelection.roof",
  ROOF_DIMENSIONS_TITLE: "roofDimensions.title",
  ROOF_DIMENSIONS_SUBTITLE: "roofDimensions.subtitle",
  ROOF_DIMENSIONS_NEXT_LABEL: "roofDimensions.nextLabel",
  ROOF_DIMENSIONS_BACK_LABEL: "roofDimensions.backLabel",
  ROOF_DIMENSIONS_PROTRUSIONS_TITLE: "roofDimensions.protrusions.title",
  ROOF_DIMENSIONS_PROTRUSIONS_ADD: "roofDimensions.protrusions.add",
  ROOF_DIMENSIONS_PROTRUSIONS_PROMPT: "roofDimensions.protrusions.prompt",
  ROOF_DIMENSIONS_PROTRUSIONS_ADD_ANOTHER:
    "roofDimensions.protrusions.addAnother",
  ROOF_DIMENSIONS_PROTRUSIONS_REMOVE: "roofDimensions.protrusions.remove",
  TILE_SELECTION_TITLE: "tileSelection.title",
  TILE_SELECTION_SUBTITLE: "tileSelection.subtitle",
  TILE_SELECTION_BACK_LABEL: "tileSelection.backLabel",
  TILE_SELECTION_CATEGORIES_CONCRETE: "tileSelection.categories.concrete",
  TILE_SELECTION_CATEGORIES_CLAY: "tileSelection.categories.clay",
  TILE_SELECTION_CATEGORIES_METAL: "tileSelection.categories.metal",
  TILE_SELECTION_COLOR: "tileSelection.color",
  TILE_SELECTION_COLORS: "tileSelection.colors",
  TILE_SELECTION_EMPTY: "tileSelection.empty",
  VARIANT_SELECTION_TITLE: "variantSelection.title",
  VARIANT_SELECTION_SUBTITLE: "variantSelection.subtitle",
  VARIANT_SELECTION_BACK_LABEL: "variantSelection.backLabel",
  TILE_OPTIONS_TITLE: "tileOptions.title",
  TILE_OPTIONS_SUBTITLE: "tileOptions.subtitle",
  TILE_OPTIONS_NEXT_LABEL: "tileOptions.nextLabel",
  TILE_OPTIONS_BACK_LABEL: "tileOptions.backLabel",
  TILE_OPTIONS_VERGE_TITLE: "tileOptions.verge.title",
  TILE_OPTIONS_VERGE_HELP: "tileOptions.verge.help",
  TILE_OPTIONS_VERGE_NONE_LABEL: "tileOptions.verge.noneLabel",
  TILE_OPTIONS_RIDGE_TITLE: "tileOptions.ridge.title",
  TILE_OPTIONS_RIDGE_HELP: "tileOptions.ridge.help",
  TILE_OPTIONS_VENTILATION_HOOD_TITLE: "tileOptions.ventilationHood.title",
  TILE_OPTIONS_VENTILATION_HOOD_HELP: "tileOptions.ventilationHood.help",
  TILE_OPTIONS_VENTILATION_HOOD_NONE_LABEL:
    "tileOptions.ventilationHood.noneLabel",
  UNDERLAY_SELECTION_TITLE: "underlaySelection.title",
  UNDERLAY_SELECTION_SUBTITLE: "underlaySelection.subtitle",
  UNDERLAY_SELECTION_NEXT_LABEL: "underlaySelection.nextLabel",
  UNDERLAY_SELECTION_BACK_LABEL: "underlaySelection.backLabel",
  UNDERLAY_SELECTION_EMPTY: "underlaySelection.empty",
  GUTTERING_TITLE: "guttering.title",
  GUTTERING_SUBTITLE: "guttering.subtitle",
  GUTTERING_NEXT_LABEL: "guttering.nextLabel",
  GUTTERING_BACK_LABEL: "guttering.backLabel",
  GUTTERING_SKIP_LABEL: "guttering.skipLabel",
  GUTTERING_GUTTER_TITLE: "guttering.gutter.title",
  GUTTERING_GUTTER_VARIANT_TITLE: "guttering.gutterVariant.title",
  GUTTERING_GUTTER_HOOK_TITLE: "guttering.gutterHook.title",
  GUTTERING_DOWN_PIPE_TITLE: "guttering.downPipe.title",
  GUTTERING_DOWN_PIPE_CONNECTORS_TITLE: "guttering.downPipeConnectors.title",
  RESULTS_TITLE: "results.title",
  RESULTS_SUBTITLE: "results.subtitle",
  RESULTS_AREA_LABEL: "results.areaLabel",
  RESULTS_BACK_LABEL: "results.backLabel",
  RESULTS_START_OVER_LABEL: "results.startOverLabel",
  RESULTS_DOWNLOAD_PDF_LABEL: "results.downloadPdfLabel",
  RESULTS_CATEGORIES_TITLES: "results.categories.tiles",
  RESULTS_CATEGORIES_FIXINGS: "results.categories.fixings",
  RESULTS_CATEGORIES_SEALING: "results.categories.sealing",
  RESULTS_CATEGORIES_VENTILATION: "results.categories.ventilation",
  RESULTS_CATEGORIES_ACCESSORIES: "results.categories.accessories",
  RESULTS_TABLE_TITLE: "results.table.title",
  RESULTS_TABLE_PACK_SIZE: "results.table.packSize",
  RESULTS_TABLE_EXTERNAL_PRODUCT_CODE: "results.table.externalProductCode",
  RESULTS_TABLE_QUANTITY: "results.table.quantity",
  RESULTS_TABLE_REMOVE: "results.table.remove",
  RESULTS_ALERTS_QUANTITIES_NOT_UPDATED_TITLE:
    "results.alerts.quantities.not.updated.title",
  RESULTS_ALERTS_QUANTITIES_NOT_UPDATED_TEXT:
    "results.alerts.quantities.not.updated.text",
  RESULTS_ALERTS_QUANTITIES_UPDATED_TITLE:
    "results.alerts.quantities.updated.title",
  RESULTS_ALERTS_QUANTITIES_UPDATED_TEXT:
    "results.alerts.quantities.updated.text",
  RESULTS_ALERTS_QUANTITIES_RESET_BUTTON:
    "results.alerts.quantities.reset.button",
  RESULTS_ALERTS_NEED_TO_KNOW_TITLE: "results.alerts.needToKnow.title",
  RESULTS_ALERTS_NEED_TO_KNOW_TEXT: "results.alerts.needToKnow.text",
  RESULTS_EMAIL_TITLE: "results.email.title",
  RESULTS_EMAIL_HELP: "results.email.help",
  RESULTS_EDITED_PRODUCTS_TITLE: "results.edited.products.title",
  PDF_REPORT_EXTRAS_SECTION_TITLE: "pdf.report.extras.section.title",
  VALIDATION_ERRORS_FIELD_REQUIRED: "validation.errors.fieldRequired",
  VALIDATION_ERRORS_POSITIVE_NUMBERS_ONLY:
    "validation.errors.positiveNumberOnly",
  VALIDATION_ERRORS_LENGTH_RANGE: "validation.errors.lengthRange",
  VALIDATION_PITCH_RANGE: "validation.errors.pitchRange",
  CALCULATOR_NOBB_LABEL: "calculator.nobb.label"
};
