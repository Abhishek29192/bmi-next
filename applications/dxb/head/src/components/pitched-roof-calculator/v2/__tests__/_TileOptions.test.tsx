import { FormContext } from "@bmi/components";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import React from "react";
import { MicroCopy } from "../../helpers/microCopy";
import { createProduct } from "../../helpers/products";
import en from "../../samples/copy/en.json";
import {
  RidgeOption,
  Tile,
  VentilationHood,
  VergeOption,
  VergeVariant
} from "../../types/v2";
import TileOptions from "../_TileOptions";

const tileVariant = createProduct<Tile>({
  color: "red",
  category: "category",
  width: 20,
  brokenBond: false,
  maxBattenSpacing: 35,
  eaveGauge: 23,
  ridgeSpacing: 5,
  minBattenSpacing: 18,
  height: 30,
  packSize: 20
});

const ridgeTile = createProduct<RidgeOption>({
  name: "Ridge tile",
  externalProductCode: "ridge_tile",
  length: 30
});

const leftVergeTile = createProduct<VergeVariant>({
  name: "left verge",
  externalProductCode: "left_verge",
  width: 10,
  code: "left_verge"
});

const rightVergeTile = createProduct<VergeVariant>({
  name: "right verge",
  externalProductCode: "right_verge",
  width: 10,
  code: "right_verge"
});

const verge: VergeOption = {
  right: rightVergeTile,
  left: leftVergeTile
};

const ventilationHood = createProduct<VentilationHood>({
  name: "Ventilation hood",
  externalProductCode: "ventilation_hood"
});

const pushEvent = jest.fn();
jest.mock("../../helpers/analytics", () => {
  const actual = jest.requireActual("../../helpers/analytics");
  return {
    ...actual,
    useAnalyticsContext: () => pushEvent
  };
});

afterEach(() => {
  jest.clearAllMocks();
});

describe("PitchedRoofCalculator TileOptions component", () => {
  it("renders with no options", () => {
    render(
      <MicroCopy.Provider values={en}>
        <FormContext.Provider
          value={{
            updateFormState: jest.fn(),
            hasBeenSubmitted: false,
            submitButtonDisabled: false,
            values: {}
          }}
        >
          <TileOptions
            variant={{
              ...tileVariant,
              vergeOption: undefined,
              ridgeOptions: [],
              ventilationHoodOptions: []
            }}
          />
        </FormContext.Provider>
      </MicroCopy.Provider>
    );

    expect(
      screen.queryByText("MC: tileOptions.verge.title")
    ).not.toBeInTheDocument();
    expect(
      screen.queryByText("MC: tileOptions.ridge.title")
    ).not.toBeInTheDocument();
    expect(
      screen.queryByText("MC: tileOptions.ventilationHood.title")
    ).not.toBeInTheDocument();
  });

  it("selects ridge tile by default", () => {
    const updateFormState = jest.fn();

    render(
      <MicroCopy.Provider values={en}>
        <FormContext.Provider
          value={{
            updateFormState,
            hasBeenSubmitted: false,
            submitButtonDisabled: false,
            values: {}
          }}
        >
          <TileOptions
            variant={{
              ...tileVariant,
              ridgeOptions: [ridgeTile],
              vergeOptions: [],
              ventilationHoodOptions: []
            }}
          />
        </FormContext.Provider>
      </MicroCopy.Provider>
    );

    expect(updateFormState).toBeCalledWith({
      ridge: ridgeTile.externalProductCode
    });
  });

  it("renders correctly with selected ventilation hood", () => {
    const updateFormState = jest.fn();

    render(
      <MicroCopy.Provider values={en}>
        <FormContext.Provider
          value={{
            updateFormState,
            hasBeenSubmitted: false,
            submitButtonDisabled: false,
            values: {}
          }}
        >
          <TileOptions
            variant={{
              ...tileVariant,
              ridgeOptions: [],
              vergeOptions: [],
              ventilationHoodOptions: [ventilationHood]
            }}
            selections={{ ventilationHoods: [ventilationHood] }}
          />
        </FormContext.Provider>
      </MicroCopy.Provider>
    );
    expect(updateFormState).toHaveBeenCalledWith(
      {
        ventilation: [ventilationHood.externalProductCode]
      },
      //second argument is an empty object because updateFormState calls with no errors
      {}
    );
  });

  it("calls pushEvent", async () => {
    render(
      <MicroCopy.Provider values={en}>
        <FormContext.Provider
          value={{
            updateFormState: jest.fn(),
            hasBeenSubmitted: false,
            submitButtonDisabled: false,
            values: {}
          }}
        >
          <TileOptions
            selections={undefined}
            variant={{
              ...tileVariant,
              vergeOption: verge,
              ridgeOptions: [
                ridgeTile,
                {
                  ...createProduct<RidgeOption>({
                    name: "Second ridge",
                    length: 25
                  })
                }
              ],
              ventilationHoodOptions: [ventilationHood]
            }}
          />
        </FormContext.Provider>
      </MicroCopy.Provider>
    );

    fireEvent.click(screen.getByText(verge.left.name));
    waitFor(() =>
      expect(pushEvent).toBeCalledWith({
        event: "dxb.button_click",
        id: "rc-options-accessories",
        label: verge.left.name,
        action: "selected"
      })
    );

    fireEvent.click(screen.getByText("MC: tileOptions.verge.noneLabel"));
    waitFor(() =>
      expect(pushEvent).toBeCalledWith({
        event: "dxb.button_click",
        id: "rc-options-accessories",
        label: "MC: tileOptions.verge.noneLabel",
        action: "selected"
      })
    );

    fireEvent.click(screen.getByText(ridgeTile.name));
    waitFor(() =>
      expect(pushEvent).toBeCalledWith({
        event: "dxb.button_click",
        id: "rc-options-accessories",
        label: ridgeTile.name,
        action: "selected"
      })
    );

    fireEvent.click(screen.getByText(ventilationHood.name));
    waitFor(() =>
      expect(pushEvent).toBeCalledWith({
        event: "dxb.button_click",
        id: "rc-options-accessories",
        label: `${ventilationHood.name} - MC: calculator.nobb.label: ${ventilationHood.externalProductCode}`,
        action: "selected"
      })
    );
  });

  it("selects 'none' option for ventilation hoods", () => {
    const updateFormState = jest.fn();

    render(
      <MicroCopy.Provider values={en}>
        <FormContext.Provider
          value={{
            updateFormState: updateFormState,
            hasBeenSubmitted: false,
            submitButtonDisabled: false,
            values: {}
          }}
        >
          <TileOptions
            selections={{ ventilationHoods: "none" }}
            variant={{
              ...tileVariant,
              vergeOption: undefined,
              ridgeOptions: [],
              ventilationHoodOptions: [ventilationHood]
            }}
          />
        </FormContext.Provider>
      </MicroCopy.Provider>
    );

    waitFor(() =>
      expect(updateFormState).toBeCalledWith({ ventilation: ["none"] }, {})
    );
  });

  it("selects 'none' option for verge tiles", () => {
    const updateFormState = jest.fn();

    render(
      <MicroCopy.Provider values={en}>
        <FormContext.Provider
          value={{
            updateFormState: updateFormState,
            hasBeenSubmitted: false,
            submitButtonDisabled: false,
            values: {}
          }}
        >
          <TileOptions
            selections={{ verge: "none", ventilationHoods: [] }}
            variant={{
              ...tileVariant,
              vergeOption: verge,
              ridgeOptions: [],
              ventilationHoodOptions: []
            }}
          />
        </FormContext.Provider>
      </MicroCopy.Provider>
    );

    waitFor(() =>
      expect(updateFormState).toBeCalledWith({ verge: "none" }, {})
    );
  });
});
