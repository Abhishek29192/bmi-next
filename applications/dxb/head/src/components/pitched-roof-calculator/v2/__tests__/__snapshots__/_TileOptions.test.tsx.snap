// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PitchedRoofCalculator TileOptions component renders correctly 1`] = `
<div>
  <div>
    <div
      class="FieldContainer"
    >
      <div
        class="head"
      >
        <h6
          class="MuiTypography-root Typography MuiTypography-h6"
        >
          MC: tileOptions.verge.title
        </h6>
        <p
          class="MuiTypography-root help Typography MuiTypography-body2"
        >
          MC: tileOptions.verge.help
        </p>
      </div>
      <div
        class="CardRadioGroup"
      >
        <div
          class="MuiGrid-root Grid MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-2"
          >
            <div
              class="CardInput item"
            >
              <label>
                <input
                  class="input"
                  name="verge"
                  type="radio"
                  value="Verge Metal Flush"
                />
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root ToggleCard card"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiPaper-root MuiCard-root body MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <div
                      class="image"
                      style="background-image: url(https://bmipimngprodtfe.azureedge.net/sys-master-hybris-media/h77/hdc/8975277129758/Product-Hero-Small-Desktop-Tablet-44134186-Icopal-Takshingel-type-S-Kullsortjpg);"
                    />
                    <h6
                      class="MuiTypography-root title Typography MuiTypography-h6"
                    >
                      Verge Metal Flush
                    </h6>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </label>
            </div>
          </div>
          <div
            class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-2"
          >
            <div
              class="CardInput item"
            >
              <label>
                <input
                  class="input"
                  name="verge"
                  type="radio"
                  value="Verge Tile"
                />
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root ToggleCard card"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiPaper-root MuiCard-root body MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <div
                      class="image"
                      style="background-image: url(https://bmipimngprodtfe.azureedge.net/sys-master-hybris-media/ha8/hf4/8975277850654/Product-Hero-Small-Desktop-Tablet-44134160-Icopal-Takshingel-type-S-Teglrodjpg);"
                    />
                    <h6
                      class="MuiTypography-root title Typography MuiTypography-h6"
                    >
                      Verge Tile
                    </h6>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </label>
            </div>
          </div>
          <div
            class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-2"
          >
            <div
              class="CardInput item"
            >
              <label>
                <input
                  class="input"
                  name="verge"
                  type="radio"
                  value="none"
                />
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root ToggleCard card"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiPaper-root MuiCard-root body MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <h6
                      class="MuiTypography-root title title--only Typography MuiTypography-h6"
                    >
                      MC: tileOptions.verge.noneLabel
                    </h6>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="FieldContainer"
    >
      <div
        class="head"
      >
        <h6
          class="MuiTypography-root Typography MuiTypography-h6"
        >
          MC: tileOptions.ridge.title
        </h6>
        <p
          class="MuiTypography-root help Typography MuiTypography-body2"
        >
          MC: tileOptions.ridge.help
        </p>
      </div>
      <div
        class="CardRadioGroup"
      >
        <div
          class="MuiGrid-root Grid MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-2"
          >
            <div
              class="CardInput item"
            >
              <label>
                <input
                  class="input"
                  name="ridge"
                  type="radio"
                  value="46035761"
                />
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root ToggleCard card"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiPaper-root MuiCard-root body MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <div
                      class="image"
                      style="background-image: url(https://bmipimngprodtfe.azureedge.net/sys-master-hybris-media/h77/hdc/8975277129758/Product-Hero-Small-Desktop-Tablet-44134186-Icopal-Takshingel-type-S-Kullsortjpg);"
                    />
                    <h6
                      class="MuiTypography-root title Typography MuiTypography-h6"
                    >
                      Zanda_Protector_ridge tile black
                    </h6>
                    <p
                      class="MuiTypography-root paragraph Typography MuiTypography-body1"
                    >
                      MC: calculator.nobb.label
                      :
                       
                      <strong>
                        46035761
                      </strong>
                    </p>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </label>
            </div>
          </div>
          <div
            class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-2"
          >
            <div
              class="CardInput item"
            >
              <label>
                <input
                  class="input"
                  name="ridge"
                  type="radio"
                  value="46035001"
                />
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root ToggleCard card"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiPaper-root MuiCard-root body MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <div
                      class="image"
                      style="background-image: url(https://bmipimngprodtfe.azureedge.net/sys-master-hybris-media/h77/hdc/8975277129758/Product-Hero-Small-Desktop-Tablet-44134186-Icopal-Takshingel-type-S-Kullsortjpg);"
                    />
                    <h6
                      class="MuiTypography-root title Typography MuiTypography-h6"
                    >
                      Zanda_Protector_ridge sample metal flush black
                    </h6>
                    <p
                      class="MuiTypography-root paragraph Typography MuiTypography-body1"
                    >
                      MC: calculator.nobb.label
                      :
                       
                      <strong>
                        46035001
                      </strong>
                    </p>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="FieldContainer"
    >
      <div
        class="head"
      >
        <h6
          class="MuiTypography-root Typography MuiTypography-h6"
        >
          MC: tileOptions.ventilationHood.title
        </h6>
        <p
          class="MuiTypography-root help Typography MuiTypography-body2"
        >
          MC: tileOptions.ventilationHood.help
        </p>
      </div>
      <div
        class="CardCheckboxGroup"
      >
        <div
          class="MuiGrid-root Grid MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-2"
          >
            <div
              class="CardInput item"
            >
              <label>
                <input
                  class="input"
                  name="ventilation"
                  type="checkbox"
                  value="100456781"
                />
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root ToggleCard card"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiPaper-root MuiCard-root body MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <div
                      class="image"
                      style="background-image: url(https://bmipimngprodtfe.azureedge.net/sys-master-hybris-media/ha8/hf4/8975277850654/Product-Hero-Small-Desktop-Tablet-44134160-Icopal-Takshingel-type-S-Teglrodjpg);"
                    />
                    <h6
                      class="MuiTypography-root title Typography MuiTypography-h6"
                    >
                      Ventilation Hood 1
                    </h6>
                    <p
                      class="MuiTypography-root paragraph Typography MuiTypography-body1"
                    >
                      MC: calculator.nobb.label
                      :
                       
                      <strong>
                        100456781
                      </strong>
                    </p>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </label>
            </div>
          </div>
          <div
            class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-2"
          >
            <div
              class="CardInput item"
            >
              <label>
                <input
                  class="input"
                  name="ventilation"
                  type="checkbox"
                  value="100456782"
                />
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root ToggleCard card"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiPaper-root MuiCard-root body MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <div
                      class="image"
                      style="background-image: url(https://bmipimngprodtfe.azureedge.net/sys-master-hybris-media/ha8/hf4/8975277850654/Product-Hero-Small-Desktop-Tablet-44134160-Icopal-Takshingel-type-S-Teglrodjpg);"
                    />
                    <h6
                      class="MuiTypography-root title Typography MuiTypography-h6"
                    >
                      Ventilation Hood 2
                    </h6>
                    <p
                      class="MuiTypography-root paragraph Typography MuiTypography-body1"
                    >
                      MC: calculator.nobb.label
                      :
                       
                      <strong>
                        100456782
                      </strong>
                    </p>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </label>
            </div>
          </div>
          <div
            class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-2"
          >
            <div
              class="CardInput"
            >
              <label>
                <input
                  class="input"
                  type="checkbox"
                  value="none"
                />
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root ToggleCard card"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiPaper-root MuiCard-root body MuiPaper-elevation1 MuiPaper-rounded"
                  >
                    <h6
                      class="MuiTypography-root title title--only Typography MuiTypography-h6"
                    >
                      MC: tileOptions.ventilationHood.noneLabel
                    </h6>
                  </div>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PitchedRoofCalculator TileOptions component renders with no options 1`] = `
<div>
  <div />
</div>
`;
